compiler/src/compiler/ast_to_hir.rs

```diff
@@ -312,24 +312,31 @@ impl<'a> Context<'a> {
 
     fn lower_text(&mut self, id: Option<ast::Id>, text: &Text) -> hir::Id {
         // TODO: Convert parts to text
-        let concatenate = self.push(
+        let builtin_text_concatenate = self.push(
             None,
             Expression::Builtin(BuiltinFunction::TextConcatenate),
             None,
         );
-        let mut result = self.push(id, Expression::Text("".to_string()), None);
-        for part in &text.0 {
-            let part_id = self.compile_single(part);
-            result = self.push(
-                None,
-                Expression::Call {
-                    function: concatenate.clone(),
-                    arguments: vec![result.clone(), part_id],
-                },
-                None,
-            );
-        }
-        result
+
+        let compiled_parts = text
+            .0
+            .iter()
+            .map(|part| self.compile_single(part))
+            .collect_vec();
+
+        compiled_parts
+            .into_iter()
```

Comment:
Can we merge these to avoid the intermediate `Vec`?

```suggestion
        text
            .0
            .iter()
            .map(|part| self.compile_single(part))
```

Reply:
That doesn't work for me because `two closures require unique access to '*self' at the same time`.
There might be a solution I'm not familiar with, but I didn't find one on the internet.